### PR TITLE
KAFKA-5588: Remove deprecated new-consumer option for tools

### PR DESCRIPTION
**Pull request by ppatierno**
Removed the deprecated "--new-consumer" option for all consumer based tools.
This option was already deprecated in the previous version so it can be removed in the coming new 2.0.0 major version.

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by viktorsomogyi — COMMENTED** [review]
Reviewed where do we use the new-consumer option. It seems that inside the ducktapes we still use them. Do you think those usages should be removed from there too?
https://github.com/apache/kafka/blob/d99f4a0ffa65ae3a654b043547f114a16fbe4905/tests/kafkatest/services/console_consumer.py#L195
https://github.com/apache/kafka/blob/d99f4a0ffa65ae3a654b043547f114a16fbe4905/tests/kafkatest/services/kafka/kafka.py#L587
https://github.com/apache/kafka/blob/d99f4a0ffa65ae3a654b043547f114a16fbe4905/tests/kafkatest/services/kafka/kafka.py#L614
https://github.com/apache/kafka/blob/d99f4a0ffa65ae3a654b043547f114a16fbe4905/tests/kafkatest/services/performance/consumer_performance.py#L48
https://github.com/apache/kafka/blob/d99f4a0ffa65ae3a654b043547f114a16fbe4905/tests/kafkatest/services/performance/consumer_performance.py#L109

**Comment by ppatierno** [comment]
@viktorsomogyi I was thinking the same thing but first waiting for a review by some committer.

**Review by vahidhashemian — COMMENTED** [review]
Thank you for the patch.

Would it be possible to also add a test case for both `ConsumerGroupCommand` and `ConsoleConsumer` that verifies using the `--new-consumer` option causes an error?

I think the upgrade document also needs to be updated to mention this change.

**Comment by ppatierno** [comment]
@vahidhashemian thanks for the feedback.
Regarding `ConsoleConsumer`, the `ConsumerConfig` class uses the `tryParse` method which just prints the help and exits if a wrong option is specified. It's different from the `ConsumerPerformace` where an exception is raised in this case (so I use that as "expected" for the unit test). Any opinion on that?
Regarding the `ConsumerGroupCommand`, we have the `ConsumerGroupCommandTest` class which is just base class for other tests but it doesn't have unit tests aboud configu itself.
I was thinking to not modify the current `ConsumerGroupCommandTest` in the `kafka.admin` package but adding a `ConsumerGroupCommandTest` in the `kafka.tools` package alongside `ConsoleConsumerTest` and `ConsumerPerformanceTest`. Wdyt ?
Regarding the upgrade doc I'll update that, thanks!

**Comment by vahidhashemian** [comment]
@ppatierno, do you think for `ConsoleConsumerTest` we can do something like in `shouldExitOnInvalidConfigWithAutoOffsetResetAndConflictingFromBeginningOldConsumer`? Also, for consumer groups, maybe something similar to `testDescribeWithMultipleSubActions` in `DescribeConsumerGroupTest`?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR. Please search the *.py code too as it has a few mentions of `--new-consumer`.

**Comment by ppatierno** [comment]
@viktorsomogyi @ijuma related to the Python files as far as I can see the `--new-consumer` option is needed for Kafka version <= 0.10.0.0 and there is a check for that in the console_consumer.py and consumer_performance.py files.
The same doesn't happen in the kafka.py.
Instead of removing the option from there I think that such a check should be added to kafka.py as well.
Wdyt ?

**Comment by ijuma** [comment]
@ppatierno Sounds good.

**Comment by ppatierno** [comment]
@vahidhashemian thanks for the suggestions, I didn't notice that `Exit` class provided a way for changing the exit behaviour. I added the unit tests which make sense about unrecognized `--new-consumer` option; I added the test for all the possible consumer group classes (describe, list, delete and reset) for consistency because they are different mutual options for the same tool.

@viktorsomogyi @ijuma I updated the Python tests as proposed.

**Review by vahidhashemian — APPROVED** [review]
Thanks for addressing my comment. LGTM.

**Review by viktorsomogyi — APPROVED** [review]
Thanks for addressing my comments too.

**Comment by ppatierno** [comment]
@vahidhashemian @viktorsomogyi thanks for your review and for accepting the changes.
@ijuma have you time to review and maybe merged by tomorrow ? Tomorrow is the deadline as far as I understood from Rajini emal : 

> Please ensure that all KIPs for 2.0.0 have been merged by Tuesday the 5th of June. Any remaining KIPs will be moved to the next release

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala
Comment (ijuma) [inline core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala]: This is not needed given the expected exception annotation. Same for other similar cases.

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala
Comment (ijuma) [inline core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala]: Nit: why are we not importing `OptionException`?

**Review by ijuma — COMMENTED** [review]
Just some minor nits.

File: core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala
Comment (ijuma) [inline core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala]: This comment seems redundant given the code.

File: docs/upgrade.html
Comment (ijuma) [inline docs/upgrade.html]: We should make it clear that we haven't changed the behaviour, `-new-consumer` was already a no-op.

File: core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala
Comment (ppatierno) [reply core/src/test/scala/unit/kafka/tools/ConsoleConsumerTest.scala]: Yeah, I am going to remove it from all the other places in the same test scala class

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala
Comment (omkreddy) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala]: Do we really need  tests for  checking unrecognized  option "--new-consumer" ?
maybe one round of manual testing should be sufficient.  maintaining these tests in repo may not be worth.

**Comment by ppatierno** [comment]
@ijuma thanks for the review. I addressed the change requests.

File: core/src/main/scala/kafka/tools/ConsumerPerformance.scala
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ConsumerPerformance.scala]: These seem to be unused?

Comment (vahidhashemian) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala]: They are used in the added unit tests in [`ConsumerPerformanceTest.scala`](https://github.com/apache/kafka/pull/5097/files#diff-7e297244dfeb67e4a86f095470b6fa92).

Comment (ppatierno) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala]: Yes exactly. They are there just for testing that without the - - new-consumer option all depends on the bootstrap servers list and zookeeper connection string and that the provided value to the configuration are what we expect.

File: core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:19]: With this import, `joptsimple.` can be dropped from the expected exception class of the first method below.
Also, no need for the `fail` statement there (similar to @ijuma's earlier comment). Not your code, but it would be great to clean them up :)

Comment (ppatierno) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:19]: Yes it makes sense. I have done the same for the `testConfigUsingNewConsumerUnrecognizedOption` in the `ConsumerPerformanceTest` class.

Comment (vahidhashemian) [reply core/src/test/scala/unit/kafka/admin/DeleteConsumerGroupsTest.scala:19]: Perfect. Thanks!

File: core/src/main/scala/kafka/tools/ConsumerPerformance.scala
Comment (ijuma) [reply core/src/main/scala/kafka/tools/ConsumerPerformance.scala]: I see. This is a bit odd and I'd rather not do that. I will push a commit with minor fixes.

**Review by ijuma — APPROVED** [review]
LGTM. I agree that we probably don't want to keep the tests for --new-consumer for long, but we can include them for now and remove them in one or two release cycles.

**Comment by ijuma** [comment]
Test failure is unrelated, merging to trunk.

**Pull request merged**
